### PR TITLE
#240: fix calculate average_backlog_size

### PR DESCRIPTION
**Pull request by Yegorov**
@yegor256 merge [this PR](https://github.com/zerocracy/fbe/pull/60) before

### CODE REVIEW

**Comment by Yegorov** [comment]
@Suban05 can you review it, please?

**Review by Suban05 — CHANGES_REQUESTED** [review]
@Yegorov looks good, but the tests failed

**Review by Suban05 — APPROVED** [review]

**Comment by Yegorov** [comment]
> @Yegorov looks good, but the tests failed

@Suban05 yes, updated fbe version, now tests are successful

File: judges/quality-of-service/quality-of-service.rb [hunk 1]
```diff
@@ -73,11 +73,13 @@
   # Average issues
   issues = []
   Fbe.unmask_repos.each do |repo|
-    Fbe.octo.repository(repo).then do |json|
-      issues << json[:open_issues]
+    (f.since.utc.to_date..Time.now.utc.to_date).each do |date|
+      issues << Fbe.octo.search_issues(
+        "repo:#{repo} type:issue created:#{date.iso8601[0..9]}..#{date.iso8601[0..9]}"
+      )[:total_count]
```
Comment (yegor256) [inline judges/quality-of-service/quality-of-service.rb:79]: @Yegorov I'm not sure this is the right logic. You count the number of opened issues. However, if all of them were opened and immediately closed, the backlog size is zero, while your counter will give big numbers. WDYT?

Comment (Yegorov) [reply judges/quality-of-service/quality-of-service.rb:79]: > However, if all of them were opened and immediately closed, the backlog size is zero

@yegor256 I search only created issues by day, it is not check open state.  
This query:
```ruby
"repo:#{repo} type:issue created:#{date.iso8601[0..9]}..#{date.iso8601[0..9]}"
```
not has [`state:open`](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests#search-by-open-or-closed-state) and return all issues created by specified day

Comment (yegor256) [reply judges/quality-of-service/quality-of-service.rb:79]: @Yegorov let's say I have three issues created on 2024-05-06 and all of them were closed the same day. What will be the size of the backlog at that day?

Comment (Yegorov) [reply judges/quality-of-service/quality-of-service.rb:79]: @yegor256 three issues.

Comment (yegor256) [reply judges/quality-of-service/quality-of-service.rb:79]: @Yegorov what if I create 3 issues on 2024-05-06 and then 4 issues on 2024-05-07, and then close them all on 2024-05-08 -- what will be backlog sizes on these three days?

Comment (Yegorov) [reply judges/quality-of-service/quality-of-service.rb:79]: @yegor256 
|date|count|
|------------|---|
| 2024-05-06 | 3 |
| 2024-05-07 | 4 |
| 2024-05-08 | 0 |

`average_backlog_size = (3 + 4 + 0)/3 = 2.33`

Comment (yegor256) [reply judges/quality-of-service/quality-of-service.rb:79]: @Yegorov exactly. However, the right answer is this:

|date|count|
|------------|---|
| 2024-05-06 | 3 |
| 2024-05-07 | 7 |
| 2024-05-08 | 0 |

`average_backlog_size = (3 + 7 + 0)/3 = 3.33`

Comment (Yegorov) [reply judges/quality-of-service/quality-of-service.rb]: @yegor256 fixed, please check

File: judges/quality-of-service/quality-of-service.rb [hunk 2]
```diff
@@ -73,11 +73,17 @@
...
   issues = []
   Fbe.unmask_repos.each do |repo|
-    Fbe.octo.repository(repo).then do |json|
-      issues << json[:open_issues]
+    (f.since.utc.to_date..Time.now.utc.to_date).each do |date|
+      count = 0
+      Fbe.octo.search_issues(
+        "repo:#{repo} type:issue created:#{f.since.utc.to_date.iso8601[0..9]}..#{date.iso8601[0..9]}"
+      )[:items].each do |item|
+        count += 1 if item[:closed_at].nil? || (!item[:closed_at].nil? && item[:closed_at].utc.to_date >= date)
```
Comment (yegor256) [inline judges/quality-of-service/quality-of-service.rb:81]: @Yegorov I believe, you can write it shorter:

```
count += 1 if item[:closed_at].nil? || item[:closed_at].utc.to_date >= date
```

Comment (Yegorov) [reply judges/quality-of-service/quality-of-service.rb]: @yegor256 cool! Thanks, fixed!

**Comment by yegor256** [comment]
@Yegorov thanks!

**Comment by 0crat** [comment]
@Suban05 Thanks for your code review! Here's your reward breakdown: +15 base points, -10 for having only 2 comments (fewer than 6), and -30 for self-review. This totals 5 points, which is our minimum reward. Your [running balance](https://zerocracy.github.io/judges-action/zerocracy-vitals.html) is now +342. Keep up the good work and remember, more detailed reviews with 6+ comments can earn you higher bonuses! 💻👍

**Comment by 0crat** [comment]
@yegor256 Hey there! 👋 Great job on the review! You've snagged +5 points this time. Here's the breakdown: +15 for showing up, but we had to dock 10 points because there were only 3 comments (the policy suggests aiming for at least 6). Keep in mind, more comments could boost your score next time - up to 20 points! Your current balance sits at [-553](https://zerocracy.github.io/judges-action/zerocracy-vitals.html). Let's aim for more detailed reviews to rack up those points! 💪

**Comment by 0crat** [comment]
@Yegorov Hey there, rockstar coder! 🚀 Awesome job on your recent contribution! You've racked up a sweet +29 points: 20 for showing up and an extra 9 for those 193 [hits-of-code](https://www.yegor256.com/2014/11/14/hits-of-code.html) you cranked out. That's some solid work, just shy of the 200 HOC mark where we start deducting points. Keep the code flowing, but remember, [quality matters](https://www.yegor256.com/2018/03/06/speed-vs-quality.html) too! Your total score is now sitting pretty at [+240](https://zerocracy.github.io/judges-action/zerocracy-vitals.html). Keep up the great work and don't forget to engage with those code reviews – they can boost your points even higher!

**Pull request merged**